RustAdapter/RustAdapter.cpp

```diff
@@ -0,0 +1,178 @@
+{
+  return m_impl->Deinitialize(shell);
+}
+
+string
+WPEFramework::Plugin::RustAdapter::Information() const
+{
+  return m_impl->Information();
+}
+
+void
+WPEFramework::Plugin::RustAdapter::AddRef() const
+{
+  m_refcount++;
+}
+
+uint32_t
+WPEFramework::Plugin::RustAdapter::Release() const
+{
+  uint32_t n = m_refcount.fetch_sub(1);
+  if (n == 1) {
+    delete this;
+    return Core::ERROR_DESTRUCTION_SUCCEEDED;
+  }
+
+  return Core::ERROR_NONE;
+}
+
+#if JSON_RPC_CONTEXT
+WPEFramework::Core::ProxyType<WPEFramework::Core::JSONRPC::Message>
+WPEFramework::Plugin::RustAdapter::Invoke(
+  const WPEFramework::Core::JSONRPC::Context &ctx,
+  const WPEFramework::Core::JSONRPC::Message &req)
+{
+  return m_impl->Invoke(ctx, req);
+}
+#else
+WPEFramework::Core::ProxyType<WPEFramework::Core::JSONRPC::Message> 
+  WPEFramework::Plugin::RustAdapter::Invoke(
+    const string& token, const uint32_t channelId, const WPEFramework::Core::JSONRPC::Message& req)
+{
+  return m_impl->Invoke(token, channelId, req);
+}
+#endif
+
+void
+WPEFramework::Plugin::RustAdapter::Activate(
+  WPEFramework::PluginHost::IShell *shell)
+{
+  m_impl->Activate(shell);
+}
+
+void
+WPEFramework::Plugin::RustAdapter::Deactivate()
+{
+  m_impl->Deactivate();
+}
+
+bool
+WPEFramework::Plugin::RustAdapter::Attach(PluginHost::Channel &channel)
+{
+  return m_impl->Attach(channel);
+}
+
+void
+WPEFramework::Plugin::RustAdapter::Detach(PluginHost::Channel &channel)
+{
+  m_impl->Detach(channel);
+}
+
+WPEFramework::Core::ProxyType<WPEFramework::Core::JSON::IElement>
+WPEFramework::Plugin::RustAdapter::Inbound(const string &identifier)
+{
+  return m_impl->Inbound(identifier);
+}
+
+WPEFramework::Core::ProxyType<WPEFramework::Core::JSON::IElement>
+WPEFramework::Plugin::RustAdapter::Inbound(const uint32_t id,
+    const Core::ProxyType<Core::JSON::IElement> &element)
+{
+  return m_impl->Inbound(id, element);
+}
+
+std::string
+WPEFramework::Plugin::RustAdapter::GetLibraryPathOrName(
+  const std::string &lib_name,
+  const std::string &callsign)
+{
+  std::stringstream shared_library_name;
+  if (!lib_name.empty())
+    shared_library_name << lib_name;
+  else {
+    shared_library_name << "lib";
+    for (char c : callsign)
+      shared_library_name << static_cast<char>(std::tolower(c));
+    shared_library_name << ".so";
+  }
+  return shared_library_name.str();
+}
+
+std::string
+WPEFramework::Plugin::RustAdapter::GetAuthToken(WPEFramework::PluginHost::IShell* service, const std::string &callsign)
+{
+  std::string auth_token;
+
+  char buff[256];
+  memset(buff, 0, sizeof(buff));
+
+  int n = snprintf(buff, sizeof(buff), "plugin://%s", callsign.c_str());
+  buff[255] = '\0';
+
+  auto auth = service->QueryInterfaceByCallsign<WPEFramework::PluginHost::IAuthenticate>("SecurityAgent");
+  if (auth != nullptr) {
+    string encoded;
+    if (auth->CreateToken(n, (const uint8_t*) buff, encoded) == WPEFramework::Core::ERROR_NONE) {
+      return encoded;
+    }
+    auth->Release();
```

Comment:
please replace `return encoded;` with`auth_token = encoded;`

Reply:
I overlooked. Thanks for pointing.. its done